pkg/controller/petset/pet.go

```diff
@@ -216,6 +220,7 @@ func (p *apiServerPetClient) Update(pet *pcb, expectedPet *pcb) (updateErr error
 		if p, getErr = pc.Get(pod.Name); getErr != nil {
 			return getErr
 		}
+		time.Sleep(retrySleepTime)
```

Comment:
Why? if this fails we'll just enqueue into a delay queue which will impose exponential build up of delay. time.sleep is generally gross because it hangs tests.